codegen/test/rtc/CMakeLists.txt

```diff
@@ -1,6 +1,21 @@
 
 find_package(hip)
+if(INSTANCES_ONLY)
+	set(CGT_TARGETS ${DEFAULT_GPU_TARGETS})
+else()
+	set(CGT_TARGETS ${GPU_TARGETS})
+endif()
+
+list(REMOVE_ITEM CGT_TARGETS gfx900 gfx906 gfx1030 gfx1100 gfx1101 gfx1102 gfx1103 gfx1200 gfx1201)
+set(offload_targets)
+foreach(target IN LISTS CGT_TARGETS)
+	string(APPEND offload_targets "--offload-arch=${target} ")
+endforeach()
+
 file(GLOB RTC_SOURCES CONFIGURE_DEPENDS src/*.cpp)
+foreach(rtc_source IN LISTS RTC_SOURCES)
+	set_source_files_properties(${rtc_source} PROPERTIES COMPILE_FLAGS ${offload_targets})
```

Comment:
These source files do not have GPU code, so a GPU target should not be set. This is a host-only API that will call to the compiler to compile GPU code at runtime(in the future this will call to hiprtc). At runtime it will compile for the current GPU(regardless of the flags passed at cmake).

Reply:
Yes, I removed the gpu targets.